lib/knock/objects.rb

```diff
@@ -137,6 +153,145 @@ def get_messages(collection:, id:, options: {})
+      # @return [Hash] The preference set (if it exists)
+      def get_preferences(collection:, id:, preference_set: DEFAULT_PREFERENCE_SET_ID)
+        endpoint = "/v1/objects/#{collection}/#{id}/preferences/#{preference_set}"
+
+        request = get_request(
+          auth: true,
+          path: endpoint
+        )
+
+        execute_request(request: request)
+      end
+
+      # Sets multiple preferences at once for the preference set.
+      #
+      # @param [String] collection The collection the object is in
+      # @param [String] id The object id
+      # @param [String] preference_set The preference set ID (defaults to `default`)
+      # @param [Hash] channel_types The channel_types hash to set
+      # @param [Hash] workflows The workflows hash to set
+      # @param [Hash] categories The categories hash to set
+      #
+      # @return [Hash] The preference set
+      def set_preferences(
+        collection:,
+        id:,
+        channel_types: nil,
+        workflows: nil,
+        categories: nil,
+        preference_set: DEFAULT_PREFERENCE_SET_ID
+      )
+        endpoint = "/v1/objects/#{collection}/#{id}/preferences/#{preference_set}"
+
+        request = put_request(
+          auth: true,
+          path: endpoint,
+          body: {
+            channel_types: channel_types,
+            workflows: workflows,
+            categories: categories
+          }
+        )
+
+        execute_request(request: request)
+      end
+
+      # Sets preferences for the given channel type
+      #
+      # @param [String] collection The collection the object is in
+      # @param [String] id The object id
+      # @param [String] preference_set The preference set ID (defaults to `default`)
+      # @param [String] channel_type The channel type to set
+      # @param [Bool] setting Whether the channel type is enabled or not
+      #
+      # @return [Hash] The preference set
+      def set_channel_type_preferences(collection:, id:, channel_type:, setting:, preference_set: DEFAULT_PREFERENCE_SET_ID)
+        endpoint = "/v1/objects/#{collection}/#{id}/preferences/#{preference_set}/channel_types/#{channel_type}"
+
+        request = put_request(
+          auth: true,
+          path: endpoint,
+          body: {subscribed: setting}
+        )
+
+        execute_request(request: request)
+      end
+
+      # Sets preferences for the given workflow
+      #
+      # @param [String] collection The collection the object is in
+      # @param [String] id The object id
+      # @param [String] preference_set The preference set ID (defaults to `default`)
+      # @param [String] workflow The workflow to set preferences for
+      # @param [Bool | Hash] setting Either a boolean to indicate if the type is enabled
+      #  or a hash containing channel types and settings
+      #
+      # @return [Hash] The preference set
+      def set_workflow_preferences(collection:, id:, workflow:, setting:, preference_set: DEFAULT_PREFERENCE_SET_ID)
```

Comment:
`DEFAULT_PREFERENCE_SET_ID` is not defined on this file (it blows up when running)

Reply:
🤦 will fix!